src/provider/model/ContextChannel.ts

```diff
@@ -86,3 +86,38 @@ export class SystemContextChannel extends ContextChannelBase {
         };
     }
 }
+
+export class AppContextChannel extends ContextChannelBase {
+    public readonly type!: 'app';
+    public readonly name: string;
+
+    private _context: Context | null;
+
+    public constructor(id: ChannelId, name: string) {
+        super(id, 'app');
+
+        this.name = name;
+
+        this._context = null;
+    }
+
+    public getStoredContext(): Context | null {
```

Comment:
is there a specific reason why we don't use a getter for this?

Reply:
Just vague intuition that it felt more appropriate, given the other methods on the class and that it's not quite as straightforward as a variable being read.

Have given changing it at try. No strong feeling, happy to go either way on this